azure-pipelines.yml

```diff
@@ -44,6 +44,24 @@ variables:
   value: y
 
 stages:
+- stage: Analysis
+  jobs:
+  - job:
+    displayName: "Semgrep"
+    continueOnError: true
+    pool:
+      vmImage: ubuntu-latest
+    steps:
+    - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
I'm not sure if it will break CI triggerred job. Because when it is CI triggerred, it looks like no step.
It's better to add checkout step explicitly.